rest/config.rst

```diff
@@ -0,0 +1,57 @@
+Config Endpoints
+================
+
+These endpoints facilitate access and modification of the :ref:`configuration
+<config>` in a granular way. Config sent to the endpoints must be in the same
+format as returned by the corresponding GET request. When posting the
+configuration succeeds, the posted configuration is immediately applied, except
+for changes that require a restart. Query :ref:`rest-config-insync` to check if
+a restart is required.
+
+For all endpoints supporting ``PATCH``, it takes the existing config and
+unmarshals the given JSON object on top of it. This means all child objects will
+replace the existing objects, not extend them. For example for
+``RawListenAddresses` in options, which is an array of strings, sending
+``{RawListenAddresses: ["tcp://10.0.0.2"]`` will replace all existing listen
+addresses.
+
+.. note::
+   Changing config in quick succession can be problematic. The changes may not
+   be applied in order, which might change the outcome. It is advicabable to
```

Comment:
We hold some sort of config lock on the API side, and we wait for the config change to commit (right?) so I think this should really be fine and not require a warning? I mean if this is the case it feels like a bug...

The problem we do have is that if you do two concurrent changes there's no way to know which gets the lock and which one has to wait, so the order is undefined -- but that's because they were concurrent, if you just do your config changes serially I think it should be OK at any rate

Reply:
Indeed we do have that lock. Mentioning this, I wasn't really thinking about that lock enough and rechecked if I missed it somewhere: https://github.com/syncthing/syncthing/pull/7053. I removed the note alltogether because I think it's common sense that you shouldn't do config changes that might conflict concurrently.